src/views/WeatherForecast/WeatherForecast.jsx

```diff
@@ -1,14 +1,63 @@
 import React from 'react';
 
+import styles from './WeatherForecast.module.scss';
 import { dataService } from '../../services/dataService';
+import { CURRENT_CITY_ID } from '../../constants/constants';
 
-class WeatherForecast extends React.Component {
-  render() {
-    return <div>Weather Forecast content</div>;
+import CityForecast from '../../layouts/CityForecast/CityForecast';
+import DetailedForecast from '../../layouts/DetailedForecast/DetailedForecast';
+
+class WeatherForecast extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dataIsReturned: false,
+      cityId: CURRENT_CITY_ID,
+      cityInfo: null,
+      cityForecast: null,
+      dailyCityForecast: null,
+      hourlyCityForecast: null
+    };
+    this.getData = this.getData.bind(this);
   }
 
   componentDidMount() {
-    dataService.getForecastToken();
+    this.getData();
+  }
+
+  async getData() {
+    try {
+      await dataService.getForecastToken();
+
+      const { cityForecast, dailyCityForecast, hourlyCityForecast, cityInfo } =
+        await dataService.getFullForecast(this.state.cityId);
+
+      this.setState({
+        cityForecast,
+        dailyCityForecast,
+        hourlyCityForecast,
+        cityInfo,
+        dataIsReturned: true
```

Comment:
isDataReturned

Reply:
you absolutely right. It will be better. Thanks